scripts/2.TCGA-MLexample_Pathway.py

```diff
@@ -0,0 +1,535 @@
+    MATCH (node:{})-[:PARTICIPATES_{}]-(gene)
+    WHERE node.identifier = "{}"
+    RETURN
+      gene.identifier as entrez_gene_id,
+      gene.name as gene_symbol
+    ORDER BY gene_symbol
+    '''.format(node_type, particip, path_id)
+    return query
+
+
+# In[6]:
+
+query_r = pathway_query(REACTOMEPATHWAY, node_type='Pathway', particip='GpPW')
+query_go_a = pathway_query(GOPATHWAY_A)
+query_go_b = pathway_query(GOPATHWAY_B)
+queries = [query_r, query_go_a, query_go_b]
+
+
+# In[7]:
+
+driver = GraphDatabase.driver("bolt://neo4j.het.io")
+full_results_df = pd.DataFrame()
+with driver.session() as session:
+    for query in queries:
+        result = session.run(query)
+        result_df = pd.DataFrame((x.values() for x in result), columns=result.keys())
+        full_results_df = full_results_df.append(result_df, ignore_index=True)
+classifier_genes_df = full_results_df.drop_duplicates().sort_values('gene_symbol').reset_index(drop=True)
+classifier_genes_df['entrez_gene_id'] = classifier_genes_df['entrez_gene_id'].astype('str')
+
+
+# In[8]:
+
+# Here are the genes that participate in the Hippo signaling pathway
+classifier_genes_df
+
+
+# In[9]:
+
+# Parameter Sweep for Hyperparameters
+n_feature_kept = 8000
+param_fixed = {
+    'loss': 'log',
+    'penalty': 'elasticnet',
+}
+param_grid = {
+    'alpha': [10 ** x for x in range(-6, 1)],
+    'l1_ratio': [0, 0.05, 0.1, 0.2, 0.5, 0.8, 0.9, 0.95, 1],
+}
+
+
+# *Here is some [documentation](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html) regarding the classifier and hyperparameters*
+
+# ## Load Data
+
+# In[10]:
+
+if not os.path.exists('data'):
+    os.makedirs('data')
+
+
+# In[11]:
+
+url_to_path = {
+    # X matrix
+    'https://ndownloader.figshare.com/files/5864859':
+        os.path.join('data', 'expression-matrix.tsv.bz2'),
+    # Y Matrix
+    'https://ndownloader.figshare.com/files/5864859':
+        os.path.join('data', 'mutation-matrix.tsv.bz2'),
+    # Clinical matrix
+    'https://ndownloader.figshare.com/files/5864859':
+        os.path.join('data', 'samples.tsv')
+}
+
+for url, path in url_to_path.items():
+    if not os.path.exists(path):
+        urllib.request.urlretrieve(url, path)
+
+
+# In[12]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('data', 'expression-matrix.tsv.bz2')\nX = pd.read_table(path, index_col=0)")
+
+
+# In[13]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('data', 'mutation-matrix.tsv.bz2')\nY = pd.read_table(path, index_col=0)")
+
+
+# In[14]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('data', 'samples.tsv')\nclinical = pd.read_table(path, index_col=0)")
+
+
+# In[15]:
+
+clinical.tail(5)
+
+
+# In[16]:
+
+# Subset the Y matrix to only the genes to be classified
+y_full = Y[classifier_genes_df['entrez_gene_id']]
+
+
+# In[17]:
+
+y_full.columns = classifier_genes_df['gene_symbol']
+y_full['disease'] = clinical['disease']
+
+
+# In[18]:
```

Comment:
colors represent different mutations - which is hard to display effectively in a legend for this particular plot (since there are 35 genes to display). Do you have any recommendations here? I am leaning towards deleting this plot and adding an extra column to the seaborn heatmap of total samples